app/src/main/java/com/google/android/sambadocumentsprovider/browsing/NetworkBrowser.java

```diff
@@ -58,7 +52,19 @@ public AsyncTask getServersAsync(OnTaskFinishedCallback<List<String>> callback)
   }
 
   private List<String> getServers() throws BrowsingException {
-    return mMasterProvider.getServers();
+    List<String> servers = null;
+
+    try {
+      servers = mMasterProvider.getServers();
+    } catch (BrowsingException e) {
```

Comment:
Not sure if this is right. Doesn't master browser return 0 server w/o throwing an exception if either side is set to SMB2+?